Correct widgets example (#59)
jtpio/jupyterlab-extension-examples: widget-tracker/widgets/README.md

# Widgets - Adding new Elements to the Main Window

In this example you will learn how to add a new tab to JupyterLab.
Visible elements such as tabs and notebooks are represented by widgets in the [PhosphorJS](http://phosphorjs.github.io/phosphor/api/widgets/globals.html)
library that is the basis of the JupyterLab application. It is the fundamental brick of
any visual component in the JupyterLab interface.

![Custom Tab](preview.png)

## A Basic Tab

The base widget class can be imported with:

```ts
// src/index.ts#L10-L10

import { Menu, Widget } from '@phosphor/widgets';
```

It requires to add the library as package dependency:

```bash
jlpm add @phosphor/widgets
```

A Widget can be added to the main area through the [JupyterLab
shell](http://jupyterlab.github.io/jupyterlab/application/classes/labshell.html).

> Inside of the `activate` function, you can obtained it through the `shell` attribute

Reviewer: ```suggestion
Inside of the `activate` function, you can obtain it through the `shell` attribute
```

of the `app` object:

```ts
// src/index.ts#L24-L24

const { commands, shell } = app;
```

> Then the widget can be inserted through the `add` method, like in the command defined

Reviewer: ```suggestion
Then the widget can be inserted by calling the `add` method, like in the command defined
```

in this example:

<!-- prettier-ignore-start -->
```ts
// src/index.ts#L30-L33

execute: () => {
  const widget = new ExampleView();
  shell.add(widget, 'main');
}
```
<!-- prettier-ignore-end -->

The custom widget `ExampleView` is inherited from the base class `Widget`. In this
case, no specific behavior is defined for the widget. Only some properties are set:

- `addClass`: Add a CSS class to allow widget styling
- `id`: id of the widget's DOM node - it is mandatory to be set to be included in JupyterLab
- `title.label`: The widget tab title
- `title.closable`: Allow the widget tab to be closed

```ts
// src/index.ts#L47-L55

class ExampleView extends Widget {
  constructor() {
    super();
    this.addClass('jp-example-view');
    this.id = 'widgets-example';
    this.title.label = 'Example View';
    this.title.closable = true;
  }
}
```

> You can associate style propreties to the custom CSS class in the file

Reviewer: ```suggestion
You can associate style properties to the custom CSS class in the file
```
